src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1697,6 +1704,13 @@ def _public_key_bytes(
         # OpenSSH + OpenSSH
         if format is serialization.PublicFormat.OpenSSH:
             if encoding is serialization.Encoding.OpenSSH:
+                if isinstance(key, dsa.DSAPublicKey):
+                    warnings.warn(
+                        "SSH DSA key support is deprecated and will be "
+                        "removed in a future release",
+                        utils.DeprecatedIn40,
+                        stacklevel=3,
+                    )
                 return ssh.serialize_ssh_public_key(key)
```

Comment:
Why doesn't this have a leading underscore :-/

Reply:
Yeah, not great. We never documented it but named it wrong.